generated/src/aws-cpp-sdk-s3-crt/source/S3CrtClient.cpp

```diff
@@ -713,7 +713,21 @@ void S3CrtClient::InitCommonCrtRequestOption(CrtRequestCallbackUserData *userDat
 
 static void CopyObjectRequestShutdownCallback(void *user_data)
 {
+  if(!user_data)
+  {
+    AWS_LOGSTREAM_ERROR("CopyObject", "user data passed is NULL ");
+    return;
+  }
   auto *userData = static_cast<S3CrtClient::CrtRequestCallbackUserData*>(user_data);
+  //log into monitor 
+  if(!HttpResponseOutcome(userData->response).IsSuccess())
```

Comment:
you should move the line 

`S3Crt::Model::CopyObjectOutcome outcome(userData->s3CrtClient->GenerateXmlOutcome(userData->response));`

before this if check, run this check on it. then return it below. it avoids constructing two outcome objects.

effectively this really just means move the logic between creating the outcome and then moving it into the response handler

Reply:
ack